app/react/Markdown/components/specs/EntityData.spec.tsx

```diff
@@ -0,0 +1,121 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from 'react';
+import { ReactWrapper } from 'enzyme';
+import Immutable from 'immutable';
+import { renderConnectedMount } from 'app/Templates/specs/utils/renderConnected';
+
+import { EntityData } from '../EntityData';
+
+describe('EntityData Markdown', () => {
+  let component: ReactWrapper<
+    Readonly<{}> & Readonly<{ children?: React.ReactNode }>,
+    Readonly<{}>,
+    React.Component<{}, {}, any>
+  >;
+  let consoleLogSpy: jasmine.Spy;
+
+  beforeEach(() => {
+    consoleLogSpy = jasmine.createSpy('consoleLogSpy');
+    spyOn(console, 'log').and.callFake(consoleLogSpy);
+  });
+
+  const render = (innerComponent: any) => {
+    const state = {
+      entityView: {
+        entity: Immutable.fromJS({
+          template: 't1',
+          title: 'Entity 1',
+          creationDate: 1234,
+          metadata: {
+            description: [{ value: 'A long description' }],
+            date: [{ value: 237600000 }],
+            main_image: [{ value: 'https://www.google.com' }],
+          },
+        }),
+      },
+      templates: Immutable.fromJS([
+        {
+          _id: 't1',
+          commonProperties: [{ name: 'title', label: 'Title' }],
+          properties: [
+            { name: 'description', type: 'text' },
+            { name: 'date', type: 'date' },
+            { name: 'main_image', label: 'Main Image', type: 'image' },
+          ],
+        },
+      ]),
+      thesauris: Immutable.fromJS([{}]),
+      translations: Immutable.fromJS([
+        {
+          locale: 'en',
+          contexts: [
+            {
+              id: 't1',
+              values: { Title: 'Title translated', 'Main Image': 'Main Image translated' },
+            },
+          ],
+        },
+      ]),
+      settings: { collection: Immutable.fromJS({ newNameGeneration: true }) },
+      inlineEdit: Immutable.fromJS({ inlineEdit: false }),
+      locale: 'en',
+    };
+
+    component = renderConnectedMount(() => innerComponent, state);
+  };
+
+  describe('root properties', () => {
+    it('should print title and root dates from root of entity', () => {
+      render(<EntityData value="title" />);
+      expect(component.html()).toBe('Entity 1');
+
+      render(<EntityData value="creationDate" />);
```

Comment:
Can I suggest the naming convention `<EntityData print="title" />`. Alternatively `display`, `show`, `get`. `render`.

Reply:
The issue is that this component can print both a value or a property name.  That is why `value` and `propertyName`  where chosen.  We can go with `printValue` or `printPropertyName`, but we need to disambiguate from both.  What would you with that taken into account?

Reply:
I see. What do you think of `valueOf='title'`and `labelOf='title'`?

Reply:
Well, `valueOf` is a reserved word, so I changed it to:
`value-of`

Since internally we adhere to camel case, internally it is called `propValueOf`.  Not nice, but there was no other workaround to keep the names consistent.